lib/restful_resource/rails_validations.rb

```diff
@@ -15,6 +15,20 @@ def put(id, data: {}, **)
         new(result)
       end
 
+      def patch(id, data: {}, **)
+        super
+      rescue HttpClient::UnprocessableEntity => e
+        errors = parse_json(e.response.body)
+        result = nil
+        result = if errors.is_a?(Hash) && errors.key?('errors')
+                   data.merge(errors)
+                 else
+                   data.merge(errors: errors)
+                 end
+        result = result.merge(id: id)
+        new(result)
```

Comment:
This block looks exactly as the one in `put`. Maybe it would be worth extracting this to a private function and reuse in both laces?